code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm

```diff
@@ -45,21 +71,54 @@
 
 /datum/action/xeno_action/toggle_bomb/action_activate()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
-	if(X.ammo.type == /datum/ammo/xeno/boiler_gas)
-		X.ammo = GLOB.ammo_list[/datum/ammo/xeno/boiler_gas/corrosive]
-		to_chat(X, span_notice("We will now fire corrosive acid. This is lethal!"))
+	var/list/spit_types = X.xeno_caste.spit_types
+	var/found_pos = spit_types.Find(X.ammo?.type)
+	if(!found_pos)
+		X.ammo = GLOB.ammo_list[spit_types[1]]
 	else
-		X.ammo = GLOB.ammo_list[/datum/ammo/xeno/boiler_gas]
-		to_chat(X, span_notice("We will now fire neurotoxic gas. This is nonlethal."))
+		X.ammo = GLOB.ammo_list[spit_types[(found_pos%length(spit_types))+1]]	//Loop around if we would exceed the length
+	var/datum/ammo/xeno/boiler_gas/BG = X.ammo
```

Comment:
It is advisable to use words for names instead of acronyms. Existing code suffers from this, but future code is good to diverge.

Reply:
Fair in this case, though disagree in some others with very widely used acronyms(T / O / M / C and so on)